reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -334,10 +336,13 @@ default Context putNonNull(Object key, @Nullable Object valueOrNull) {
 	default Context putAll(Context other) {
 		if (other.isEmpty()) return this;
 
-		return other.stream()
-		            .reduce(this,
-				            (c, e) -> c.put(e.getKey(), e.getValue()),
-				            (c1, c2) -> { throw new UnsupportedOperationException("Context.putAll should not use a parallelized stream");}
-		            );
+		if (other instanceof AbstractContext && this instanceof AbstractContext) {
```

Comment:
I would still recommend having `AbstractContext#putAll` and avoid one `instanceof` check, since the context is a performance sensitive API

Reply:
`isEmpty()` check can also be avoided (since it does `this instanceof Context0`)

Reply:
that's only valid for the default implementation, but any external implementor of `Context` could have redefined `isEmpty()`.

Reply:
I'll reintroduce `putAll` in `AbstractContext` with the `instanceof AbstractContext` check first, then delegating to `Context` default method, avoiding double `isEmpty()` in that case.